internal/product/accessor.go

```diff
@@ -154,32 +143,45 @@ func (p *postgresProductAccessor) GetProductsByVendor(
 	}
 	defer rows.Close()
 
-	res := []Product{}
+	res := []ProductVendor{}
 
 	for rows.Next() {
-		var product Product
-		if err := rows.StructScan(&product); err != nil {
+		var productVendor ProductVendor
+		if err := rows.StructScan(&productVendor); err != nil {
 			return nil, err
 		}
-		res = append(res, product)
+		res = append(res, productVendor)
 	}
 	if err := rows.Err(); err != nil {
 		return nil, err
 	}
 
-	countQuery := `SELECT COUNT(*) FROM product_vendor WHERE vendor_id=$1`
+	countQuery := `SELECT COUNT(*)
+		FROM product_vendor pv
+		JOIN price pr ON pr.product_vendor_id = pv.id
+		WHERE pr.vendor_id = $1`
```

Comment:
I just noticed this query is wrong... What if the filter is on, the countQuery would still be the unfiltered count 🤦‍♂️

Reply:
hwemm nice catch edu, good point